File: Code Samples/Stack - Nikita Choudhari/Stack Implemenation using  Array/stack.py
MAX_SIZE = 1000

class Stack:
    def __init__(self):
        self.top = -1
        self.arr = [0] * MAX_SIZE

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == MAX_SIZE - 1

    def push(self, x):
        if self.isFull():
            print("Stack Overflow")
        else:
            self.top += 1
            self.arr[self.top] = x

    def pop(self):
        if self.isEmpty():
            print("Stack Underflow")
        else:
            self.top -= 1

    def peek(self):
        if self.isEmpty():
            print("Stack is Empty")
            return -1
        return self.arr[self.top]


stack = Stack()
stack.push(10)
stack.push(20)
stack.push(30)
print(stack.peek())
stack.pop()
print(stack.peek())

File: Code Samples/Stack - Nikita Choudhari/Stack Implemenation using  Array/Stack.java
public class Stack {
    final int MAX_SIZE = 1000;
    int top;
    int[] arr = new int[MAX_SIZE];

    Stack() {
        top = -1;
    }

    boolean isEmpty() {
        return (top == -1);
    }

    boolean isFull() {
        return (top == MAX_SIZE - 1);
    }

    void push(int x) {
        if (isFull()) {
            System.out.println("Stack Overflow");
        } else {
            arr[++top] = x;
        }
    }

    void pop() {
        if (isEmpty()) {
            System.out.println("Stack Underflow");
        } else {
            --top;
        }
    }

    int peek() {
        if (isEmpty()) {
            System.out.println("Stack is Empty");
            return -1;
        }
        return arr[top];
    }

    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(10);
        stack.push(20);
        stack.push(30);
        System.out.println(stack.peek());
        stack.pop();
        System.out.println(stack.peek());
    }
}
